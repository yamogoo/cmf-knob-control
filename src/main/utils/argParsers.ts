export const parseEnvArgs = (str: string, opst = { divider: ',' }): Array<string> => {
  const arr = str.split(opst.divider)
  return arr
}

export const parseVPIdEnvArgs = (str: string): Array<Array<string>> => {
  const items = parseEnvArgs(str)

  const arr: Array<Array<string>> = []

  for (let i = 0; i < items.length; i++) {
    const item = items[i]

    const args = parseEnvArgs(item, { divider: ':' })
    arr.push(args)
  }

  return arr
}
