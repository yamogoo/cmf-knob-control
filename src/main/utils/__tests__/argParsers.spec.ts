import { describe, expect, test } from 'vitest'

import { parseEnvArgs, parseVPIdEnvArgs } from '../argParsers'

describe('parseEnvArgs', () => {
  test.each(['abcd:efgh,1234:5678'])('%s should parsed to ["abcd:efgh", "1234:5678"]', (str) => {
    const args = parseEnvArgs(str)

    expect(args).toMatchObject(['abcd:efgh', '1234:5678'])
    expect(args).toMatchSnapshot()
  })

  test.each(['key:val', 'abcd:1234'])('%s should heave 2 args', (str) => {
    const args = parseEnvArgs(str, { divider: ':' })
    const length = args.length

    expect(length).toBe(2)
    expect(length).toMatchSnapshot()
  })
})

describe('parseVPIdEnvArgs', () => {
  test.each(['a:1,b:2,c:3'])('(%s) should have args length 3', (str) => {
    const args = parseVPIdEnvArgs(str)
    const length = args.length

    expect(length).toBe(3)
    expect(length).toMatchSnapshot()
  })

  test.each(['a:1,b:2,c:3'])('(%s) the "c" arg should be "3"', (str) => {
    const args = parseVPIdEnvArgs(str)

    const c = args[2][1]
    expect(c).toBe('3')
    expect(c).toMatchSnapshot()
  })

  test.each(['a,b'])('(%s) should have args length 2', (str) => {
    const args = parseVPIdEnvArgs(str)

    expect(args.length).toBe(2)
    expect(args).toMatchSnapshot()
  })
})
