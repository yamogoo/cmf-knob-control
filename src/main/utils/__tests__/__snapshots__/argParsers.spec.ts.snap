// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseEnvArgs > abcd:1234 should heave 2 args 1`] = `2`;

exports[`parseEnvArgs > abcd:efgh,1234:5678 should parsed to ["abcd:efgh", "1234:5678"] 1`] = `
[
  "abcd:efgh",
  "1234:5678",
]
`;

exports[`parseEnvArgs > key:val should heave 2 args 1`] = `2`;

exports[`parseVPIdEnvArgs > (a,b) should have args length 2 1`] = `
[
  [
    "a",
  ],
  [
    "b",
  ],
]
`;

exports[`parseVPIdEnvArgs > (a:1,b:2,c:3) should have args length 3 1`] = `3`;

exports[`parseVPIdEnvArgs > (a:1,b:2,c:3) the "c" arg should be "3" 1`] = `"3"`;
